src/subcommands/extract_slice/extract.rs

```diff
@@ -14,3 +15,13 @@ pub fn extract_slice<P1: AsRef<Path>, P2: AsRef<Path>>(
     global_state::transfer_global_state(&db_path, &output, state_root_hash)?;
     Ok(())
 }
+
+pub fn extract_slice_with_root<P1: AsRef<Path>, P2: AsRef<Path>>(
```

Comment:
We should have the same `extract_slice` function with a sort of `Identifier` parameter which would be an `enum` with either the block hash or the state root hash. Then we can easily merge the existing `extract_slice` with your new function.